pkg/objectstore/objectstore.go

```diff
@@ -0,0 +1,100 @@
+package objectstore
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/glebarez/sqlite"
+	"gorm.io/gorm"
+	"k8s.io/utils/strings/slices"
+)
+
+var ErrUnregisteredType = func(t interface{}) error { return fmt.Errorf("%T is not registered with objectstore", t) }
+
+type DB struct {
+	location        string
+	db              *gorm.DB
+	registeredTypes []string
+}
+
+type DBConfig struct {
+	Location string
+}
+
+func NewObjectStore(options ...Option) *DB {
+	config := DBConfig{
+		Location: ":memory:",
```

Comment:
You maybe want to modify this to `:memory:?busy_timeout=5000`:
> busy_timeout=5000 sets a busy handler that sleeps for a specified amount of time when a table is locked.
The handler will sleep multiple times until at least "ms" milliseconds of sleeping have accumulated.
After at least "ms" milliseconds of sleeping, the handler returns 0 which causes sqlite3_step() to return SQLITE_BUSY
We do this to prevent the error "database is locked" from being returned on parallel reads and writes.